lib/Dialect/RTL/RTLTypes.cpp

```diff
@@ -238,8 +238,8 @@ void ArrayType::print(DialectAsmPrinter &p) const {
 LogicalResult ArrayType::verifyConstructionInvariants(Location loc,
                                                       Type innerType,
                                                       size_t size) {
-  if (!isRTLValueType(innerType))
```

Comment:
This change is a separate PR (#528) but necessary for this one. I won't merge this one until #528 merges.